feat: :sparkles: Fixed typo and add new metrics. Updated grafana

a-lp opened the pull request:
Updated grafana dashboard with new pie charts

The new metrics permit to monitor the amount per transaction category and transaction group of category.

sakowicz commented:
Thank you for the contribution.

1. Could you add tests so it will be easier to manage this in the future?
2. Can we use both `actual_budget_category_tranasction_count` and `actual_budget_category_transacion_count`, so we don't break existing dashboards? We can deprecate the one with typo.

I don't have an access to PC now so I'll test it after a week.

a-lp commented:
> 1. Could you add tests so it will be easier to manage this in the future?

I've added new tests and restored the `actual_budget_category_tranasction_count` metric, keeping the one with correct name. The tests verify the correct extraction of category group names and payees names. They also test that all the metrics are exported. 

> 2. Can we use both actual_budget_category_tranasction_count and actual_budget_category_transaction_count, so we don't break existing dashboards? We can deprecate the one with typo.

Actually, in my first commit I updated the grafana dashboard in the repository to point to the correct metric, so it should work with the one with the typo fixed.

sakowicz commented:
I've tested your changes - thanks for the work!
That said, I'm not sure how useful the current setup is. The panels show amounts by category/group starting from the beginning of time, which makes it hard to spot recent trends. Maybe showing only the last 30/60/90 days or the current month would be more helpful.
Also, I'm not sure Prometheus is the best tool for aggregations like this. Curious what you think.

a-lp commented:
> I've tested your changes - thanks for the work! That said, I'm not sure how useful the current setup is. The panels show amounts by category/group starting from the beginning of time, which makes it hard to spot recent trends. Maybe showing only the last 30/60/90 days or the current month would be more helpful. Also, I'm not sure Prometheus is the best tool for aggregations like this. Curious what you think.

I agree that prometheus is not the best aggregator of informations, that's why I switched to actual-http-api in order to query the data directly to the server. For what regards the dates, they can be customized. I use the picker to choice the date range, even single months in the past.
